WalletWasabi/Blockchain/TransactionBuilding/TransactionModifierWalletExtensions.cs

```diff
@@ -280,6 +282,24 @@ public static BuildTransactionResult CpfpTransaction(this Wallet wallet, SmartTr
 		var destination = keyManager.GetNextChangeKey().GetAssumedScriptPubKey().GetDestinationAddress(network);
 		Guard.NotNull(nameof(destination), destination);
 
+		// Request the unconfirmed transaction chain so we can extract the fee paid by tx + all the ancestors still unconfirmed.
+		using var cts = new CancellationTokenSource(TimeSpan.FromSeconds(20));
+		EffectiveTransactionStatus? unconfirmedTransactionChain = null;
+		try
+		{
+			wallet.UnconfirmedTransactionChainProvider
+				.ImmediateRequestAsync(transactionToCpfp, cts.Token)
+				.GetAwaiter()
+				.GetResult();
+		}
+		catch (Exception ex)
+		{
+			Logger.LogWarning($"Error while trying to get the unconfirmed transaction chain of tx {transactionToCpfp.GetHash()}: {ex}");
+		}
+
+		var ancestorsSizeBytes = unconfirmedTransactionChain is null ? 0 : (long)Math.Ceiling(unconfirmedTransactionChain.ancestors.Sum(x => x.weight) / 4.0);
+		var feePaidByAncestorsAndTx = unconfirmedTransactionChain is null ? 0 : unconfirmedTransactionChain.ancestors.Sum(x => x.fee) + (decimal)unconfirmedTransactionChain.adjustedVsize;
```

Comment:
You are adding fees and vsizes here, those are different units and cannot be added.

Reply:
Thanks there was a real mistake here